src/librustc/diagnostics.rs

```diff
@@ -1944,6 +1944,38 @@ you want. Example:
 ```
 "##,
 
+E0492: r##"
+A borrow of a constant containing interior mutability was attempted. Erroneous
+code example:
+
+```
+#![feature(const_fn)]
+
+use std::cell::Cell;
+
+const A: Cell<usize> = Cell::new(1);
+const B: &'static Cell<usize> = &A;
+// error: cannot borrow a constant which contains interior mutability, create
+//        a static instead
+
+// or:
+struct C { a: Cell<usize> }
+
+const D: C = C { a: Cell::new(1) };
+const E: &'static Cell<usize> = &D.a; // error
+
+// or:
+const F: &'static C = &D; // error
+
+fn main() {}
+```
+
+To solve this error, either use statics or avoid using `cell` type. If you want
```

Comment:
> The issue with the statics solution is that there's no safe way to do it (statics can't have destructors), and getting the unsafe way correct depends on the exact situation.

I don't understand your point, because destructors are entirely orthogonal: types with or without destructors can have (or not have) interior mutability.

Here's a before/after example that I was thinking of:

```rust
use std::sync::atomic::{AtomicUsize, ATOMIC_USIZE_INIT};

// before:
// const A: AtomicUsize = ATOMIC_USIZE_INIT;
// static B: &'static AtomicUsize = &A;

// after:
static C: AtomicUsize = ATOMIC_USIZE_INIT;
static D: &'static AtomicUsize = &C;

fn main() {}
```